FreeRTOS-Plus/Demo/FreeRTOS_Plus_WolfSSL_FIPS_Ready_Windows_Simulator/README_wolfSSL_FIPS_ready.md

```diff
@@ -0,0 +1,91 @@
+
+Next to this folder you will see another demo folder named "FreeRTOS_Plus_WolfSSL_Windows_Simulator". The demo uses regular (non-FIPS-Ready) wolfSSL. If you compare both demos, you will notice that there are no changes to the client code, and will also notice that some additional tests run prior to your main program in this FIPS-Ready demo.
+
+This demo shows that wolfSSL FIPS Ready provides a FIPS compliant cryptographic module with minimal impact on client code. 
+
+# What does FIPS 140-2 specify?
+
+FIPS 140-2 enforces cryptographic modules to follow best practices, including:
+
+1. Removal of insecure algorithms (such as MD5 and DES)
+2. Include a default entry point
+3. Perform a Power On Self Test (POST)
+
+wolfSSL FIPS Ready fulfils these requirements. The third requirement means that the POST should run automatically whenever the application using the FIPS code starts up. For wolfSSL FIPS Ready, the POST consits of two tests: 
+
+- In-Core Integrity Check (HMAC-SHA256 over cryptographic algorithm object files)
+- Known Answer Tests (KATs)
+
+The in-core integrity check performs an HMAC-SHA256 operation over the object files included in the FIPS-compliant algorithm boundary. The cryptographic boundary is the FIPS-specific code and its related static data in the memory of the program. In the integrity check process, the calculated hash value is compared with the expexted pre-calculated value in the memory. Failure of this check means that compiled boundary code was modified after it was compiled. If either the integrity check or KAT fails, the module enters an error state.
+
+The KAT (Known Answer Tests) run algorithm test cases using pre-computed NIST test vectors, thus verifying that the algorithms are working successfully. The KAT code and test vectors are inside the cryptographic boundary and are also checked as part of the in-core integrity check.
+
+# How to build and run the Demo application
+
+By double-clicking the solution file named "FreeRTOS_Plus_WolfSSL_FIPS_Ready.sln" included in this folder, Visual Studio starts and shows you a project in its solution explorer. It is named "RTOSDemo" and provides a console application program which runs on windows. 
+
+All required settings have been set in the user_settings.h header file included in the RTOSDemo/FreeRTOS+/wolfSSL folder in the solution explorer pane. The next step is to build the RTOSDemo application:
+
+1. Build the RTOSDemo project
+2. Run the RTOSDemo.exe 
+
+You will see a console that pops up, and it shows output like the following:
+
+```
+Starting Power On Self Test
+In core integrity check error: hash = C66491A040D5B9686BAA7A75A280290D91B49...
+ERROR: -203
+Power On Self Test(Known-Answer-Test) FAILURE
+```

```

Comment:
i only got: 
In core integrity check error: hash = 6DC89FFF8455F1DCD9CF2CF5BE942CC2CDD57F226ABE07BFA673E79A88F8C9B5

Reply:
Thank you for your advice. Corrected the demo output description to match the actual demo output in c1a9bd.